monkey/monkey_island/cc/agent_event_handlers/scan_event_handler.py

```diff
@@ -55,71 +62,54 @@ def handle_tcp_scan_event(self, event: TCPScanEvent):
         except (RetrievalError, StorageError, UnknownRecordError):
             logger.exception("Unable to process tcp scan data")
 
-    def _get_target_machine(self, event: ScanEvent) -> Machine:
-        try:
-            target_machines = self._machine_repository.get_machines_by_ip(event.target)
-            return target_machines[0]
-        except UnknownRecordError:
-            machine = Machine(
-                id=self._machine_repository.get_new_id(),
-                network_interfaces=[IPv4Interface(event.target)],
-            )
-            self._machine_repository.upsert_machine(machine)
-            return machine
-
-    def _get_source_node(self, event: ScanEvent) -> Node:
-        machine = self._get_source_machine(event)
-        try:
-            node = self._node_repository.get_node_by_machine_id(machine.id)
-        except UnknownRecordError:
-            node = Node(machine_id=machine.id)
-            self._node_repository.upsert_node(node)
-        return node
-
-    def _get_source_machine(self, event: ScanEvent) -> Machine:
-        agent = self._agent_repository.get_agent_by_id(event.source)
-        return self._machine_repository.get_machine_by_id(agent.machine_id)
+    @staticmethod
+    def _get_open_ports(event: TCPScanEvent) -> List[NetworkPort]:
+        return [port for port, status in event.ports.items() if status == PortStatus.OPEN]
 
     def _update_target_machine_os(self, machine: Machine, event: PingScanEvent):
         if event.os is not None and machine.operating_system is None:
             machine.operating_system = event.os
             self._machine_repository.upsert_machine(machine)
 
-    def _update_network_services(self, target: Machine, event: TCPScanEvent):
-        network_services = {
+    @classmethod
+    def _get_tcp_connections_from_event(cls, event: TCPScanEvent) -> Sequence[SocketAddress]:
+        tcp_connections = set()
+        open_ports = cls._get_open_ports(event)
+        for open_port in open_ports:
+            socket_address = SocketAddress(ip=event.target, port=open_port)
+            tcp_connections.add(socket_address)
+
+        return tuple(tcp_connections)
+
+    @classmethod
+    def _get_network_services_from_event(cls, event: TCPScanEvent) -> NetworkServices:
+        return {
             SocketAddress(ip=event.target, port=port): NetworkService.UNKNOWN
-            for port in self._get_open_ports(event)
+            for port in cls._get_open_ports(event)
         }
-        self._machine_repository.upsert_network_services(target.id, network_services)
-
-    @staticmethod
-    def _get_open_ports(event: TCPScanEvent) -> List[NetworkPort]:
-        return [port for port, status in event.ports.items() if status == PortStatus.OPEN]
 
-    def _update_nodes(self, target_machine: Machine, event: ScanEvent):
-        src_machine = self._get_source_machine(event)
+    def _upsert_from_tcp_scan_event(
+        self,
+        event: TCPScanEvent,
```

Comment:
Sending source and target is more readable

Reply:
How?

Reply:
It's always better to have unpacked values in params. Type hints show what you're working with and method doesn't get exposed to values it doesn't need. Not a deal-breaker though